### PR TITLE
Niconico live search feature

### PR DESCRIPTION
**Pull request by yayugu**
example:

```ruby
client.search(
  '麻倉もも',
  [
    Niconico::Live::Client::SEARCH_FILTER_CLOSED,
    Niconico::Live::Client::SEARCH_FILTER_OFFICIAL,
    Niconico::Live::Client::SEARCH_FILTER_CHANNEL,
    Niconico::Live::Client::SEARCH_FILTER_HIDE_TS_EXPIRED,
  ]
)
#=>
[#<struct Niconico::Live::SearchResult
  id="lv218679185",
  title="【おまけ生放送】THE IDOLM@STER MillionRADIO",
  description=
   "「アイドルマスター ミリオンラジオ！」チャンネルでは、会員限定のおまけ生放送をお送りします！\r\n\r\n※おまけ生放送の放送時間は...">,
 #<struct Niconico::Live::SearchResult
  id="lv215593983",
  title="THE IDOLM@STER MillionRADIO",
  description=
   "GREEで好評配信中のソーシャルゲーム「アイドルマスター ミリオンライブ！」\r\n話題沸騰中の同ゲームのウェブラジオをお届けしま...">,
 #<struct Niconico::Live::SearchResult
  id="lv180901966",
  title=
   "電人☆ゲッチャ！：『フリーダムウォーズ』『CROSS†CHANNEL～For all people～』『PlayStation Plus』",
  description=
   "ニコ生公式ゲーム＆アニメ情報番組\r\n『電人☆ゲッチャ！』\r\n今週は『CROSS†CHANNEL～For all people～（クロスチャンネル フォー ...">]
```

### CODE REVIEW

**Comment by sorah** [comment]
feature is ok

can you move constants like Niconico::Live::Client::SearchFilters::ONAIR ?

**Comment by yayugu** [comment]
:)

File: lib/niconico/live/client.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+require 'niconico/live/client/search_result'
+require 'niconico/live/client/search_filters.rb'
```
Comment (sorah) [inline lib/niconico/live/client.rb]: .rb

**Comment by yayugu** [comment]
oops

**Comment by sorah** [comment]
:sparkles:

**Pull request merged**